keel/quotations/admin.py

```diff
@@ -16,7 +28,38 @@ class QuotationAdmin(admin.ModelAdmin):
 
 
 class QuotationMilestioneAdmin(CustomBaseModelAdmin):
-    pass
+    
+    def save_model(self, request, obj, form, change):
+        id = str(obj.pk)
+        customer = obj.quotation.user
+        
+        try:
+            case = Case.objects.get(user=customer)
+        except Exception as e:
+            logger.error('ERROR: QUOTATIONS:QuotationMilestioneAdmin ' + str(e)) 
+            return super().save_model(request, obj, form, change)
+        
+        payment_manager = PaymentManager()
+        data = {
```

Comment:
No need for nested dictionary, you could directly write 

order_items = {
"quotation_milestone":[id],
                    "plan": [],
                    "service": []
}


and use this variable in generate_payment_details function.